Source/Core/GeometryPipeline.js

```diff
@@ -1919,13 +1919,46 @@ define([
     var p0Scratch = new Cartesian3();
     var p1Scratch = new Cartesian3();
     var p2Scratch = new Cartesian3();
-    var barycentricScratch = new Cartesian3();
+    function interpolateAndPackCartesian3(i0, i1, i2, coords, sourceValues, currentValues, insertedIndex, normalize) {
```

Comment:
Since this and the below function only differ with the cartesian type and the number of components, why not generate them?

```javascript
function generateInterpolateFunction(cartesian, numbeOfComponents) {
    var p0Scratch = new cartesian();
    // ...
    return function(i0, i1, /*...*/) {
        var p0 = cartesian.fromArray(sourceValues, i0 * numComponents, p0Scratch);
        // ...
    };
}
var interpolateAndPackCartesian3 = generateInterpolateFunction(Carteisan3, 3);
```

Reply:
@mramato Does this impact performance? If so, it might be fine since this most likely only gets executed in a web worker.